tsnet/tsnet.go

```diff
@@ -323,6 +323,17 @@ func (s *Server) doInit() {
 	}
 }
 
+// TailscaleIPs returns IPv4 and IPv6 addresses for this node. If the node
+// has not yet joined a tailnet or is otherwise unaware of its own IP addrsses,
+// an empty slice will be returned.
+func (s *Server) TailscaleIPs() (addrs []netip.Addr) {
```

Comment:
If this had the signature of returning v4 and v6 then it'd both not allocate and not risk aliasing the NetMap memory.

Reply:
Is it likely that we'll have multiple IP addresses for specific purposes? https://github.com/tailscale/tailscale/issues/3847

Maybe that would just mean changing the API (sorry, folks).

Reply:
I came here to suggest the same as Brad, not for performance but because it would be much easier to reason about. I suspect users, like me, come looking for the single Tailscale IP, and learn it has some arbitrary number of IPs in a sort order I don't understand. v4/v6 would make it clear and what I should do.

Reply:
Done.